mqr/tests/test_views.py

```diff
@@ -129,3 +130,105 @@ def test_faq_message(self, mock_get_message_details):
         )
 
         mock_get_message_details.assert_called_with("BCM_week_pre22_faq1")
+
+
+class StrataRandomization(APITestCase):
+    url = reverse("mqr_randomstrataarm")
+
+    def test_random_arm_unauthorized_user(self):
+        """
+        unauthorized user access denied
+        Returns: status code 401
+
+        """
+
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_random_arm(self):
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+
+        response = self.client.post(
+            self.url,
+            data={
+                "province": "EC",
+                "weeks_pregnant_bucket": "21-25",
+                "age_bucket": "31+",
+            },
+            format="json",
+        )
+
+        strata_arm = MqrStrata.objects.get(
+            province="EC", weeks_pregnant_bucket="21-25", age_bucket="31+"
+        )
+
+        self.assertContains(response, "random_arm")
```

Comment:
I would rather check the index and that the response contains the first item in the random list

Reply:
Done